#!/bin/bash

PATCH="firefox-unity-support.patch"

patch_firefox() {
for i in www-client/firefox/firefox-${1}.*.ebuild; do
	cat > ${PATCH} << EOF
--- a/${i}	2013-10-12 22:06:13.218057063 +0200
+++ b/${i}	2013-10-12 22:37:42.203441496 +0200
@@ -44,7 +44,8 @@
 # More URIs appended below...
 SRC_URI="\${SRC_URI}
 	http://dev.gentoo.org/~anarchy/mozilla/patchsets/\${PATCH}.tar.xz
-	http://dev.gentoo.org/~nirbheek/mozilla/patchsets/\${PATCH}.tar.xz"
+	http://dev.gentoo.org/~nirbheek/mozilla/patchsets/\${PATCH}.tar.xz
+	https://launchpad.net/ubuntu/saucy/+source/\${PN}/${1}.0+build${2}/+files/\${PN}_${1}.0+build${2}.debian.tar.gz"
 
 ASM_DEPEND=">=dev-lang/yasm-1.1"
 
@@ -143,6 +144,7 @@
 }
 
 src_prepare() {
+	epatch "\${WORKDIR}/debian/patches/unity-menubar.patch"
 	# Apply our patches
 	EPATCH_SUFFIX="patch" \\
 	EPATCH_FORCE="yes" \\
EOF
	git apply ${PATCH}
	rm ${PATCH}
done
}

patch_firefox 24 1-0ubuntu1
patch_firefox 25 3-0ubuntu0.13.10.1
cat >> www-client/firefox/Manifest << EOF
DIST firefox_24.0+build1-0ubuntu1.debian.tar.gz 273652 SHA256 4e3f88622381def717b8f9d5c39aef916faadb88ced8c458b1496f187eadc390 SHA512 73dc738d2aab5f6dbc9faf60e61b0f1795db6ce773b13551207fc26a9ccd783a92cb0ca4f0c21f95e78b4ab74f9a62dc5d0497d2d92dca55f663d59e324b520d WHIRLPOOL 9f9aa0beb7fc69a97dd9381c88db5d62f9041dfed30f8470024f98a1a1d9cd0a8483e10ed018defd3c10673b84ddc55bdbe6d2e6049992264eeab818a36ad44e
DIST firefox_25.0+build3-0ubuntu0.13.10.1.debian.tar.gz 273370 SHA256 b8cb9557ed039748a290e7d713b32fa1b24f06d559139fe5c7a009a6778f064d SHA512 8ef795fc5553fb8a1b88e01c4d91b25ea2189e5e688116b87efb5870d4acf8a88e94a4aab697f13f84833fb5b5f9e8a096740c817dd4d5ca8b104f7fb31e19eb WHIRLPOOL 6197aed278bba6f6c75dac922caaada9d7bcc0811b587fd2fa55046311156e08a89129d25b3987e1316b0ddf5540cc2e794c6cbd6f90d4bec86248217142e6e7
EOF
